Sources/NIO/BlockingIOThreadPool.swift

```diff
@@ -153,6 +155,31 @@ public final class BlockingIOThreadPool {
     }
 }
 
+public extension BlockingIOThreadPool {
+    /// Runs the submitted closure if the thread pool is still active, otherwise fails the promise.
+    /// The closure will be run on the thread pool so can do blocking work.
+    ///
+    /// - parameters:
+    ///     - eventLoop: The `EventLoop` to create the returned `EventLoopFuture` on.
+    ///     - body: The closure which performs some blocking work to be done on the thread pool.
+    /// - returns: An `EventLoopFuture` fulfilled with the result (or error) of the passed closure.
+    public func runIfActive<T>(eventLoop: EventLoop, _ body: @escaping () throws -> T) -> EventLoopFuture<T> {
```

Comment:
hmm, unconvinced as there are more things that can go wrong if we ask the user to give us a promise instead of just making one. What's the benefit of taking a promise instead of the `EventLoop`?

Reply:
The major benefit is that the user controls where the future fires. That's basically it though.

Reply:
Yup, basically isomorphic here.

Reply:
@weissi for me taking a `EventLoop` looks more like running something on the `EventLoop`, which is not the case here. Also taking an `EventLoopPromise` is more consistent with what we do for `ChannelOutboundInvoker` imho.

Reply:
also @Lukasa

Reply:
yeah, let @Lukasa be the tie breaker.

Reply:
Ok, I think we should take a promise: it's more consistent with the other interfaces.